### DISCUSSION TITLE
Need Help , please give some suggestions [Solved]

### DISCUSSION THREAD

**Discussion by soumya-shome** [msys2/MINGW-packages#7813] [General]
I am trying to do mysql connection using c++, on Code::Blocks and mingw compiler on MSYS2.
 
Upon Compliling 
```
g++.exe -Wall -fexceptions  -c "D:\Programming Languages\03. Projects\cppdatabase\main.cpp" -o obj\Release\main.o
g++.exe  -o bin\Release\cppdatabase.exe obj\Release\main.o
```
I am getting this error 

```   
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: obj\Release\main.o:main.cpp:(.text+0x13): undefined reference to `mysql_init'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: obj\Release\main.o:main.cpp:(.text+0x5d): undefined reference to `mysql_real_connect'
collect2.exe: error: ld returned 1 exit status
 ```
My Code 
```
#include<iostream>
#include <mysql/mysql.h>

using namespace std;

int main()

{
    MYSQL* conn;
    conn = mysql_init(0);
    conn = mysql_real_connect(conn, "localhost", "root", "", "db_test", 0, NULL, 0);

    if(conn)
    {

        cout <<" Connection to database successful:"<<endl;
    }
    else{
        cout<<"Connection unsuccessful"<<endl;
    }
    return 0;
}
```

I have this installed : 
```pacman -S mingw-w64-x86_64-libmariadbclient   ```

I tried different methods available online , none of them helped.

Any suggestions ??

**Comment by 1480c1** [comment]
you probably need to add the libraries to link to mariadb, try changing your final linking line to 

```bash
g++.exe  -o bin\Release\cppdatabase.exe obj\Release\main.o $(pkgconf --libs --static libmariadb)
```

**Reply by Biswa96** [reply]
Not right. It's `pkgconf --libs --static libmariadb`.

**Reply by 1480c1** [reply]
Ah, I was using manjaro and just guessed the name, it seems it has both

**Comment by Biswa96** [comment]
I think you have just forgot to link to the libmariadb.dll library. Just add the linking option `-lmariadb` to your g++ command. The resultant command will be like this `g++ -Wall test.cpp -lmariadb`.

If you have more questions you are welcomed join Gitter or Discord.

**Reply by soumya-shome** [reply]
Yes , this helped . adding ```-lmariadb``` to the other linker setting solved the problem.
Thanks a lot.
